tree-sitter-dscexpression/grammar.js

```diff
@@ -4,16 +4,17 @@ module.exports = grammar({
   rules: {
     statement: $ => choice(
       $.escapedStringLiteral,
-      $.expression,
+      $._expressionString,
       $.stringLiteral,
     ),
-    escapedStringLiteral: $ => token(seq('[[', /.*/)),
-    expression: $ => seq('[', $.function, optional($._members), ']'),
+    escapedStringLiteral: $ => token(prec(1, seq('[[', /.*?/, ']'))),
```

Comment:
Regarding the `[test] string` input, ideally, this rule should be `seq(/^\[\[/, /.*?/, /]$/)`, however, tree-sitter doesn't support the `^` and `$` assertions today. According to https://github.com/tree-sitter/tree-sitter-cli/issues/53, we need to use a custom scanner in order to check for "starting with" or "ending with" something.

See the newer issue https://github.com/tree-sitter/tree-sitter/issues/1970 for additional details.

Reply:
I saw that, but I don't think this will be a problem since the entire statement is a single line.